geps/gep-3779/index.md

```diff
@@ -42,20 +37,432 @@ Kubernetes core provides NetworkPolicies as one way to do it. Network Policies h
+
+###### Option 2
+
+Leaving an empty targetRefs. While this is theoretically possible without a breaking change, this would be another fundamental change to policy attachment (to allow policies without a targetRef).
+Additionally, this is also suffer from the same problem of being perceived as an applicable scope for both Gateways and Workloads in the namespace.
+
+This option is also inconsistent with other API fields where an empty field or absence of a it does not mean select all. See [recent comment](https://github.com/kubernetes-sigs/gateway-api/pull/3887#discussion_r2176125600) on the proposal for empty ParentRef field.
+
+###### Option 3 (Recommended)
+
+An empty pod selector. Kubernetes official docs clarify that the semantics of empty selectors are the decision of the API owner. In fact, many Kubernetes APIs (I know Service API does the opposite :/) using empty selectors as a select-all mechanism. See [NetworkPolicy podSelector](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.33/#networkpolicy-v1-networking-k8s-io), PodDisruptionBudget, ResourceQuota, and more.
+
+```yaml
+targetRefs:
+- Kind: Pod
+  Selector: {}
+```
+
+This provides a clearer separation between workloads and gateways. Gateways that happen to be Pods in the cluster, will also get selected. However they can be excluded with the right `matchExpression` if desired.
+
+Below is a pseudo example:
+
+```yaml
+targetRefs:
+- Kind: Pod
+  Selector:
+    matchExpression:
+    - { key: "purpose", operator: NotIn, values: ["gateway"] }
+```
+
+##### New EnforcementLevel Enum
+
+To mitigate the challenges L7 support across dataplanes is going to present, and to encourage policy authors to be more explicit, we introduce a new EnforcementLevel Enum.
+
+We start by only support the Network value, and Application is kept for future iteration.
+
+Note: The whole point of this Enum is to encourage explicitness, we **do not** want to start without it and default to Network if and when introduced later.
+
+```go
+
+// EnforcementLevel defines the scope at which an AuthorizationPolicy is enforced.
+//
+// There are two enforcement levels:
+//
+//   - Network: Enforces the policy at the network layer (L4), typically at
+//     network proxies or gateway dataplanes. Only supports attributes available
+//     at connection time (e.g., source identity, port). Recommended for broad,
+//     coarse-grained access controls.
+//
+//     NOTE FOR REVIEWRS -- THIS IS FOR FUTURE ENHANCEMENT
+//
+//   - Application: Enforces the policy at the application layer (L7), typically at
+//     HTTP/gRPC-aware sidecars or L7 proxies. Enables fine-grained authorization
+//     using protocol-specific attributes (e.g., HTTP paths, methods).
+//
+// This field clarifies policy intent and informs where enforcement is expected
+// to happen. It also enables implementation-specific validation and behavior.
+//
+// +kubebuilder:validation:Enum=Network;Application
```

Comment:
If only `Network` is supported, I'd think validation shouldn't allow `Application`.